src/types/facets.ts

```diff
@@ -3,5 +3,5 @@ export interface Facet {
 }
 
 export interface Facets {
-  [key: string]: Facet
+  [key: string]: Facet | Facets
 }
```

Comment:
Isn't what you want something along the lines of:

```typescript
export type Facets = { [key: string]: Facet } & { topFacets: Facets }
```

The advantage would be that you don't need to "guess" the type for keys other than `topFacets`